### PR TITLE
Fixed type abbreviation from bypassing ByRefLike rules

### PR DESCRIPTION
**Pull request by TIHan**
This should be a simple fix. Resolves https://github.com/dotnet/fsharp/issues/6133 and https://github.com/dotnet/fsharp/issues/6133.  Looks like this PR: https://github.com/dotnet/fsharp/pull/6135 was intended to do the same thing too.

### CODE REVIEW

**Comment by abelbraaksma** [comment]
> Resolves #6133 and #6133.

That's twice the same issue, did you mean to include another bug report as well? ;)

File: src/fsharp/PostInferenceChecks.fs [hunk 1]
```diff
@@ -457,7 +464,7 @@ let CheckEscapes cenv allowProtected m syntacticArgs body = (* m is a range suit
 let AccessInternalsVisibleToAsInternal thisCompPath internalsVisibleToPaths access =
     // Each internalsVisibleToPath is a compPath for the internals of some assembly.
     // Replace those by the compPath for the internals of this assembly.
-    // This makes those internals visible here, but still internal. Bug://3737
+    // This makes those internals visible here, but still internal. Bug://3737bui
```
Comment (cartermp) [inline src/fsharp/PostInferenceChecks.fs]: bui?

File: tests/fsharp/Compiler/Language/SpanTests.fs [hunk 2]
```diff
@@ -50,4 +51,55 @@ test ()
...
+    0
+            """
+            [|
+                FSharpErrorSeverity.Error, 412, (11, 17, 11, 28), "A type instantiation involves a byref type. This is not permitted by the rules of Common IL."
+                FSharpErrorSeverity.Error, 412, (19, 17, 19, 29), "A type instantiation involves a byref type. This is not permitted by the rules of Common IL."
+                FSharpErrorSeverity.Error, 412, (19, 27, 19, 29), "A type instantiation involves a byref type. This is not permitted by the rules of Common IL."
+            |]
+
+    [<Test>]
+    let ``Should fail to compile on a type abbreviation 2``() =
```
Comment (cartermp) [inline tests/fsharp/Compiler/Language/SpanTests.fs]:
```suggestion
    let ``Type abbreviation that boxes a span should fail to compile``() =
```

File: tests/fsharp/Compiler/Language/SpanTests.fs [hunk 3]
```diff
@@ -50,4 +51,55 @@ test ()
         // We expect this error until System.Reflection.Emit gets fixed for emitting `modreq` on method calls.
         // See: https://github.com/dotnet/corefx/issues/29254
         CompilerAssert.RunScript script [ "Method not found: '!0 ByRef System.ReadOnlySpan`1.get_Item(Int32)'." ]
+
+
+    [<Test>]
+    let ``Should fail to compile on a type abbreviation``() =
```
Comment (cartermp) [inline tests/fsharp/Compiler/Language/SpanTests.fs]:
```suggestion
    let ``Invalid usage of type abbreviated span should fail to compile``() =
```

File: src/fsharp/PostInferenceChecks.fs (hunk 1 above)
Comment (TIHan) [reply src/fsharp/PostInferenceChecks.fs]: bui

**Pull request merged**